build-release-tools/lib/PlatformClients.py

```diff
@@ -130,28 +138,67 @@ def provider_exist(self, atlas_version, provider):
+        if resp.ok:
+            versions = [item["version"] for item in resp.json()["versions"]]
+        else:
+            raise Exception("Failed to get box versions {0}\n{1}".format(self.box, resp.text))
+        return versions
+
+    def del_box_version(self, version):
+        """
+        Delete one box version
+        """
+        del_box_versions_url = self.generate_url("version", version)
+        resp = self.session.delete(del_box_versions_url)
+        if resp.ok:
+            print "Delete {0}/{1} successfully!".format(self.box, version)
+        else:
+            raise Exception("Failed to delete {0}/{1}\n{2}!".format(self.box, version, resp.text))
+
     def generate_url(self, purpose, atlas_version=None, provider=None):
         """
         Tool method, Generate all possible urls according to purpose
         """
         purpose_handler = {
-            "check_version": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}".format(atlas_version)]),
-            "create_version": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "versions"]),
-            "check_provider": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}/provider/{1}".format(atlas_version, provider)]),
-            "create_provider": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}/providers".format(atlas_version)]),
-            "upload_box": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}/provider/{1}".format(atlas_version, provider), "upload"]),
-            "release_box": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}".format(atlas_version, provider), "release"])
+            "version": "/".join([self.atlas_url, self.box, "version/{0}".format(atlas_version)]),
+            "create_version": "/".join([self.atlas_url, self.box, "versions"]),
+            "check_provider": "/".join([self.atlas_url, self.box, "version/{0}/provider/{1}".format(atlas_version, provider)]),
+            "create_provider": "/".join([self.atlas_url, self.box, "version/{0}/providers".format(atlas_version)]),
+            "upload_box": "/".join([self.atlas_url, self.box, "version/{0}/provider/{1}".format(atlas_version, provider), "upload"]),
+            "release_box": "/".join([self.atlas_url, self.box, "version/{0}".format(atlas_version, provider), "release"]),
+            "box": "/".join([self.atlas_url, self.box])
         }
-        return purpose_handler[purpose](atlas_version, provider)
+        return purpose_handler[purpose]
 
 class Bintray(object):
     """
     Bintray client for calling Bintray API.
     """
-    def __init__(self, creds, subject, repo, push_executable, **kwargs):
-        self._username, self._api_key = common.parse_credential_variable(creds)
+    def __init__(self, creds, subject, repo, push_executable=None, **kwargs):
+        assert creds, "creds is None!"
```

Comment:
Thanks, good comments.
I thinks that's the meaning of ```-O``` option.

simple assert cost less resource, but if assert is like this
``` 
assert value in some_expensive_function()
```
User may need an option to avoid expensive assert in prod env....

IMO "-O" provides a way to disabled unnecessary or expensive assert in some situation ,
but not the reason for don't use assert.